conan/internal/api/uploader.py

```diff
@@ -271,13 +271,36 @@ def gzopen_without_timestamps(name, fileobj, compresslevel=None):
     return t
 
 
-def compress_files(files, name, dest_dir, compresslevel=None, ref=None, recursive=False):
+def compress_files(files, name, dest_dir, conf=None, ref=None, recursive=False, compression_plugin=None):
+    tgz_path = os.path.join(dest_dir, name)
+    if compression_plugin:
+        t1 = time.time()
+        compressed_path = compression_plugin.tar_compress(
+            archive_path=os.path.join(dest_dir, COMPRESSED_PLUGIN_TAR_NAME),
```

Comment:
Can the user ignore the COMPRESSED_PLUGIN_TAR_NAME and return any other arbitrary name?

Reply:
The plugin functions will recive an absolute path to the compressed file without the extension.
Is the plugin responsibility to add the extension and return the final path (with the extension)

Users could return anything, yes. But if they do so, it won't work. 
I will add an extra validation to check a correct path has been returned.

Reply:
Why not make the user return only the extension?

Reply:
Yes, that is another aproach I thoght. Before this last refactor, the complete path was needed as the user could change the name. Now that the name is constant... it seems reasonable to return only the extension yep.
I'll update it!